python/mdvtools/auth/auth0_provider.py

```
@@ -379,37 +407,77 @@ def sync_users_to_db(self):
+                            auth_id=auth0_id
                         )
 
-            logging.info("Successfully synced users from Auth0 to the database.")
+                        # Add delay between role requests to avoid rate limiting
+                        time.sleep(0.2)  # 200ms delay between requests
+                        
+                        try:
+                            # Fetch user's roles with retry mechanism
+                            @retry_with_exponential_backoff
+                            def get_user_roles():
+                                return auth0.users.list_roles(auth0_id)
```

Ruff (0.11.9)

Function definition does not bind loop variable `auth0_id`

(B023)